### PR TITLE
Cme feature: Target button - support for all types of entities

### PR DESCRIPTION
**Pull request by RadStr**

### CODE REVIEW

File: applications/conceptual-model-editor/src/action/center-viewport-to-visual-entity.ts [hunk 1]
```diff
@@ -1,18 +1,41 @@
+import { isVisualGroup, isVisualNode, isVisualRelationship } from "@dataspecer/core-v2/visual-model";
 import { ModelGraphContextType } from "../context/model-context";
 import { UseDiagramType } from "../diagram/diagram-hook";
 import { UseNotificationServiceWriterType } from "../notification/notification-service-context";
+import { SemanticModelRelationship } from "@dataspecer/core-v2/semantic-model/concepts";
+import { isSemanticModelRelationshipUsage, SemanticModelRelationshipUsage } from "@dataspecer/core-v2/semantic-model/usage/concepts";
+import { ClassesContextType } from "../context/classes-context";
+import { isAnAttribute, temporaryDomainRangeHelper } from "../util/relationship-utils";
 
+// TODO: Falls apart if we want to represent 1 semantic entities by more visual ones, in such case we want to pass in identifiers of both semantic and the visual entity.
```
Comment (skodapetr) [inline applications/conceptual-model-editor/src/action/center-viewport-to-visual-entity.ts:10]: Move this to an issue. I do not see how this should be resolved localy.

File: applications/conceptual-model-editor/src/diagram/diagram-controller.ts [hunk 2]
```diff
@@ -456,6 +456,13 @@ const createActions = (
         focusNodeAction(reactFlow, node);
       }
     },
+    fitToView(identifiers) {
+      console.log("Diagram.fitToView", { identifiers });
```
Comment (skodapetr) [inline applications/conceptual-model-editor/src/diagram/diagram-controller.ts:460]: Do we need this log here? Seems like it was used for debug during development of  a particular feature, perhaps not needed anymore?

**Review by skodapetr — CHANGES_REQUESTED** [review]
Just two minors, before merge.

**Comment by RadStr** [comment]
Should be fine now - Removed the two lines.

On a side note, I noticed two things in the main branch:

1. Layouting doesn't work - When you were rewriting part of code you wrote if instead of else if, but that is fixed in another branch, for which I will open PR next.
2. The target button no longer works, but maybe that will be fixed by this merge

File: applications/conceptual-model-editor/src/action/center-viewport-to-visual-entity.ts (hunk 1 above)
Comment (skodapetr) [reply applications/conceptual-model-editor/src/action/center-viewport-to-visual-entity.ts]: Both, I would like to have this as an issues, as we plan to identify entities using model identifier and entity identifier.

**Comment by skodapetr** [comment]
We can merge once the conflict is resolved.

**Comment by skodapetr** [comment]
Yet, it would be best to have a way how to enforce the right match, model to model, entity to entity, in TS. Perhaps we can introduce a new type to wrap both of them together as  a way to reference an entity. We can discuss this in the issue.

**Comment by RadStr** [comment]
> Yet, it would be best to have a way how to enforce the right match, model to model, entity to entity, in TS. Perhaps we can introduce a new type to wrap both of them together as a way to reference an entity. We can discuss this in the issue.

I am not sure if we are thinking about the same issue, but I know what you mean. We can quickly discuss that tomorrow if there will be time.

**Pull request merged**